### DISCUSSION TITLE
Theia browser-only + VS Code Web Extension example source code

### DISCUSSION THREAD

**Discussion by mr-sagar-shelar** [eclipse-theia/theia#14388] [Q&A]
Hi,
I've tried to run Theia in browser only mode and able to run it locally.
Next thing which I want to try is add Language Server Protocol support to theia in browser only mode by implementing VS Code Web Extension. To do this I've created local web only plugin using VS Code Extension example

https://github.com/microsoft/vscode-extension-samples/tree/main/lsp-web-extension-sample

The extension works fine in VS Code and to add make it work with Theia I followed all instruction one by one mentioned in this docs
https://theia-ide.org/docs/authoring_vscode_extensions

But the extensions are is not working on browser only mode.

There is already a POC available for browser only mode which supports Typescript Language Support
https://theia-ide.org/docs/authoring_vscode_extensions/ 

This POC was mention here https://github.com/eclipse-theia/theia/issues/12852 and for the source code it is pointing to https://github.com/eclipsesource/theia/blob/browser-only-plugins-poc

This branch and repository is outdated I think.

Requesting community to please help me with finding source code for Theia browser only mode with VS Code Web Extension (LSP) support or how it can be achieved.

Thanks in advance.

**Comment by msujew** [comment]
Hey @mr-sagar-shelar,

The feature is currently not supported, due to considerable effort of making this work for the generic use case. The showcase for the typescript language support was a hack to show its general feasibility. I don't believe it was intended as a way to show adopters how to deploy plugins for their browser only app.

However, I believe there are people looking into this. I think @sdirix might know more.

**Reply by sdirix** [reply]
In general web extensions themselves work in Theia browser-only. However in the current state of the code base, the metadata extraction and deployment process of plugins (i.e. VS Code [web] extensions) is implemented in the Theia backend and is therefore not available in Theia browser-only.

If you are motivated, you could for now use the same approach as we did in our `poc` by:
 - extracting the metadata manually, for example by inspecting the metadata of your web extensions when loaded in the the regular Theia `browser` application
 - hard coding the extracted metadata by overwriting all kinds of services
 - placing the web extensions next to the built `browser-only` application

As a guideline you can use the [last commit](https://github.com/eclipsesource/theia/commit/cd095237db9c3e68273aef30ef0167b89fea4ffb) of the https://github.com/eclipsesource/theia/tree/browser-only-plugins-poc branch.

---

@robertjndw is currently looking into streamlining this process to offer a straightforward way of bundling web extensions with a browser-only application. However I can't give you any timeline here. After this is completed the next interesting step would be to allow to install and deploy web extensions dynamically at runtime.

@mr-sagar-shelar if you, or someone else reading this, is interested in contributing to this feature then feel free to contact me to organize this.
